File: Osp/src/main/java/de/gso/osp/persistence/model/Todo.java
package de.gso.osp.persistence.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "todo")
public class Todo {

    @Id
    @Column(unique = true, nullable = false)
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    
    private String todoString;
    
    private boolean done;
    
    private Benutzer benutzer;
    
    private Klasse klasse;
    
    private Unterricht unterricht;

	public String getTodoString() {
		return todoString;
	}

	public void setTodoString(String todoString) {
		this.todoString = todoString;
	}

	public boolean isDone() {
		return done;
	}

	public void setDone(boolean done) {
		this.done = done;
	}

    @ManyToOne
    @JoinColumn(name = "benutzer_id")
	public Benutzer getBenutzer() {
		return benutzer;
	}

	public void setBenutzer(Benutzer benutzer) {
		this.benutzer = benutzer;
	}

    @ManyToOne
    @JoinColumn(name = "klasse_id")
	public Klasse getKlasse() {
		return klasse;
	}

	public void setKlasse(Klasse klasse) {
		this.klasse = klasse;
	}
	
    @ManyToOne
    @JoinColumn(name = "unterricht_id")
	public Unterricht getUnterricht() {
		return unterricht;
	}

	public void setUnterricht(Unterricht unterricht) {
		this.unterricht = unterricht;
	}

}

File: Osp-PHP/src/Entity/Todo.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass="App\Repository\TodoRepository")
 */
class Todo
{
    /**
     * @ORM\Id()
     * @ORM\GeneratedValue()
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $todoString;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $todoKey;

    /**
     * @ORM\Column(type="boolean")
     */
    private $erledigt;

    /**
     * @ORM\ManyToOne(targetEntity="App\Entity\Benutzer", inversedBy="todos")
     * @ORM\JoinColumn(nullable=false)
     */
    private $benutzer;

    /**
     * @ORM\ManyToOne(targetEntity="App\Entity\Klasse", inversedBy="todos")
     * @ORM\JoinColumn(nullable=false)
     */
    private $klasse;

    /**
     * @ORM\ManyToOne(targetEntity="App\Entity\Fach", inversedBy="todos")
     * @ORM\JoinColumn(nullable=false)
     */
    private $fach;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTodoString(): ?string
    {
        return $this->todoString;
    }

    public function setTodoString(string $todoString): self
    {
        $this->todoString = $todoString;

        return $this;
    }

    public function getTodoKey(): ?string
    {
        return $this->todoKey;
    }

    public function setTodoKey(string $todoKey): self
    {
        $this->todoKey = $todoKey;

        return $this;
    }

    public function getErledigt(): ?bool
    {
        return $this->erledigt;
    }

    public function setErledigt(bool $erledigt): self
    {
        $this->erledigt = $erledigt;

        return $this;
    }

    public function getBenutzer(): ?Benutzer
    {
        return $this->benutzer;
    }

    public function setBenutzer(?Benutzer $benutzer): self
    {
        $this->benutzer = $benutzer;

        return $this;
    }

    public function getKlasse(): ?Klasse
    {
        return $this->klasse;
    }

    public function setKlasse(?Klasse $klasse): self
    {
        $this->klasse = $klasse;

        return $this;
    }

    public function getFach(): ?Fach
    {
        return $this->fach;
    }

    public function setFach(?Fach $fach): self
    {
        $this->fach = $fach;

        return $this;
    }
}
